src/controllers/user.ts

```diff
@@ -0,0 +1,231 @@
+import { Request, Response } from "express";
+import { PrismaClient } from '@prisma/client'
+import bcrypt from "bcrypt";
+import { signupBody,signinBody, updateUserBody } from "../Validation/ZodValidation";
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+dotenv.config();
+
+const client = new PrismaClient();
+
+const JWT_SECRET = process.env.JWT_SECRET || "defaultsecretkey"; 
+const signup = async (req: Request, res: Response) => {
+    try {
+        const body = req.body;
+        const parseResult = signupBody.safeParse(body);
+        if (!parseResult.success) {
+            return res.status(400).json({ message: "Invalid Inputs", success:false});
+        }
+        const { name, username, mobile, email, password, institute, yearOfStudy, interests, isAdmin } = parseResult.data;
+        if (!email.endsWith("@iitkgp.ac.in")) {
+            return res.status(400).json({
+                message: "Email must end with @iitkgp.ac.in",
+                success: false,
+            });
+        }
+        const existingUser = await client.user.findUnique({
+            where: { email },
+        });
+        if (existingUser) {
+            return res.status(409).json({ message: "Email already registered" ,success:false});
+        }
+        const saltRounds = 10;
+        const hashedPassword = await bcrypt.hash(password, saltRounds);
+        const newUser = await client.user.create({
+            data: {
+                name,
+                username,
+                mobile,
+                email,
+                password: hashedPassword,
+                institute,
+                yearOfStudy,
+                interests,
+                isAdmin,
+            },
+        });
+        const token = jwt.sign({ id: newUser.id }, JWT_SECRET, { expiresIn: "1h" }); 
+        return res.status(201).json({
+            message: "User registered successfully",
+            user: { id: newUser.id, email: newUser.email },
+            token,
+            success:true
+        });
+    } catch (error) {
+        console.error("Error in signup:", error);
+        return res.status(500).json({ message: "Internal Server Error",success:false });
+    }
+};
+
+ const signin = async (req: Request, res: Response) => {
+    try {
+        const body = req.body;
+        const parseResult = signinBody.safeParse(body);
+        if (!parseResult.success) {
+            return res.status(400).json({
+                message: "Invalid Inputs",
+                success:false
+            });
+        }
+        const { email, password } = parseResult.data;
+        if (!email.endsWith("@iitkgp.ac.in")) {
+            return res.status(400).json({
+                message: "Email must end with @iitkgp.ac.in",
```

Comment:
The email should end with `@kgpian.itkgp.ac.in` and I think the `message` could be `Please enter Institute email address`